_posts/2022-02-16-solidity-0.8.12-release-announcement.md

```diff
@@ -0,0 +1,37 @@
+---
+layout: post
+published: true
+title: 'Solidity 0.8.12 Release Announcement'
+date: '2022-02-16'
+author: Solidity Team
+category: Releases
+---
+
+[Solidity v0.8.12](https://github.com/ethereum/solidity/releases/tag/v0.8.12)
+improves the javascript/wasm binary and fixes several bugs.
+
+## Notable New Features
+
+### Emscripten build / solcjs
+
+The code for the javascript wrappers of the compiler has been partially
+ported to TypeScript, which also impacts some of the aspects of the npm package.
+We hope this is seamless to you but please do not hesitate to provide feedback.
+
+Furthermore, we were able to reduce the size of the javascript / emscripten
+binaries from 27 megabytes to just over 8. The reason for the large binary
+in the first place is that we include the SMT solver Z3.
+The size reduction is achieved by compressing the binary using LZ4 and
+decompressing it when it is loaded, which should be completely transparent to the user.
+We were initially worried that this might have a negative impact on people who
+switch compiler versions often or always run it in a fresh nodejs environment,
+but it turns out that the decompression is faster than the base64-decoding,
```

Comment:
I'd qualify that statement just to be safe, though I hope that it won't turn out to be slower in some weird setups or with older versions.
```suggestion
but our benchmarks show that the decompression is faster than the base64-decoding,
```